orttraining/orttraining/eager/ort_aten.cpp

```diff
@@ -499,9 +499,20 @@ at::Tensor _copy_from_and_resize(
   const auto ort_self = create_ort_value(invoker, self);
   auto ort_dst = create_ort_value(invoker, dst);
 
+  // If the size of the dst is not equal to src, re-allocate the dst tensor to
+  // match the src tensor size.
+  if (self.numel() != dst.numel()) {
+    auto& dst_ort_tensor = ort_dst.Get<onnxruntime::Tensor>();
+    onnxruntime::Tensor::InitOrtValue(dst_ort_tensor.DataType(), onnxruntime::TensorShape(self.sizes().vec()),
```

Comment:
will this cause the memory leak of original tensor's buffer

Reply:
From my reading this will not leak:

If follow call to [Tensor::InitOrtValue(..)](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/framework/tensor.cc#L75), and then [OrtValue::Init(...)](https://github.com/microsoft/onnxruntime/blob/master/include/onnxruntime/core/framework/ort_value.h#L35), we can see it is performing [reset](https://github.com/microsoft/onnxruntime/blob/master/include/onnxruntime/core/framework/ort_value.h#L36) on shared_ptr, which should release the original tensor's buffer if necessary.